### DISCUSSION TITLE
Clearing Action Dialog? Invoking backups?

### DISCUSSION THREAD

**Discussion by jcmil101** [hluk/CopyQ#1934] [Q&A]
Just discovered this great little program, I am reading thru the docs but there are some gaps in my understanding.

Whilst experimenting I added some code to the F5 dialog & having hit _save_, I'm now wondering how to remove it....
I don't see my entry in the ini's but it must be somewhere as it is still in the dropdown...?

I noticed the **Backup On Exit.ini** but cannot see how to invoke it.  I can add it as code via F6 & I can paste it as an .ini into the dir but am uncertain how to invoke it on shutdown.
Any help appreciated.

Love the ability to use regex on the copied items, I've already started building that up for my needs.  :)

**Comment by hluk** [comment]
The dropdown in the Action dialog shows only history of previously triggered commands. Whereas the save button allows you to save the action as a menu item, shortcut action or automatic command run on clipboard change. This is stored in the Command dialog accessible using F6 shortcut.

**Reply by jcmil101** [reply]
OK thanks.
I figured out how to clean the F6 dialog so it was only the F5 view which was annoying me  :D

**Reply by hluk** [reply]
I am probably going to remove the Save button from GUI completely. It is a bit obscure and does not work that well ... and I never needed it. :)

**Comment by jcmil101** [comment]
I wonder if it would be useful with a tooltip for noobs like me?
I just started looking at S&R in the F2 dialog & see no replace function, I assume I can script a multi pass button to format a bunch of entries.

In that case if using the F5 dialog for testing, I might use the _save_ function in the F5 dialog....  
........Assuming you don't already have a better way of doing it already.

Thanks for a great program Lukas. :)

**Comment by jcmil101** [comment]
Wondering if something like the **Map** function exists in the prog, I found this java snippet & tried a variation on this without success:-

```
var replacements = new Map([
        [/[ -]/g, '_'],
        [/[\r\n$#[\]]/g, '']
        ]),
```

It would solve a range of SARs for me if you have something like this already implemented.
